File: VB/WindowsApplication1/CustomDescriptor/UnboundRow.vb
Imports System
Imports System.ComponentModel

Namespace WindowsApplication1
    Public Class UnboundRow
        Inherits PropertyDescriptor

        Private _MyValue As Object
        Private ReadOnly _ValueType As Type
        Public Property MyValue() As Object
            Get
                Return _MyValue
            End Get
            Set(ByVal value As Object)
                _MyValue = value
            End Set
        End Property

        Public Sub New(ByVal name As String, ByVal valueType As Type)
            MyBase.New(name, Nothing)
            _ValueType = valueType
        End Sub


        Protected Sub New(ByVal name As String, ByVal attrs() As Attribute)
            MyBase.New(name, attrs)

        End Sub
        Protected Sub New(ByVal descr As MemberDescriptor)
            MyBase.New(descr)

        End Sub
        Protected Sub New(ByVal descr As MemberDescriptor, ByVal attrs() As Attribute)
            MyBase.New(descr, attrs)

        End Sub

        Public Overrides ReadOnly Property Category() As String
            Get
                Return "!Unbound rows"
            End Get
        End Property


        Public Overrides Function CanResetValue(ByVal component As Object) As Boolean
            Return False
        End Function

        Public Overrides ReadOnly Property ComponentType() As Type
            Get
                Return _ValueType
            End Get
        End Property

        Public Overrides Function GetValue(ByVal component As Object) As Object
            Return MyValue
        End Function

        Public Overrides ReadOnly Property IsReadOnly() As Boolean
            Get
                Return False
            End Get
        End Property

        Public Overrides ReadOnly Property PropertyType() As Type
            Get
                Return _ValueType
            End Get
        End Property

        Public Overrides Sub ResetValue(ByVal component As Object)

        End Sub

        Public Overrides Sub SetValue(ByVal component As Object, ByVal value As Object)
            MyValue = value
        End Sub

        Public Overrides Function ShouldSerializeValue(ByVal component As Object) As Boolean
            Return False
        End Function
    End Class
End Namespace

File: CS/DXApplication3/CustomDescriptor/UnboundRow.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;

namespace DXApplication3
{
    public class UnboundRow : PropertyDescriptor
    {

        private object _MyValue;
        private readonly Type _ValueType;
        public object MyValue
        {
            get { return _MyValue; }
            set { _MyValue = value; }
        }

        public UnboundRow(string name, Type valueType)
            : base(name, null)
        {
            _ValueType = valueType;
        }


        protected UnboundRow(string name, Attribute[] attrs)
            : base(name, attrs)
        {

        }
        protected UnboundRow(MemberDescriptor descr)
            : base(descr)
        {

        }
        protected UnboundRow(MemberDescriptor descr, Attribute[] attrs)
            : base(descr, attrs)
        {

        }

        public override string Category
        {
            get
            {
                return "!Unbound rows";
            }
        }


        public override bool CanResetValue(object component)
        {
            return false;
        }

        public override Type ComponentType
        {
            get { return _ValueType; }
        }

        public override object GetValue(object component)
        {
            return MyValue;
        }

        public override bool IsReadOnly
        {
            get { return false; }
        }

        public override Type PropertyType
        {
            get { return _ValueType; }
        }

        public override void ResetValue(object component)
        {

        }

        public override void SetValue(object component, object value)
        {
            MyValue = value;
        }

        public override bool ShouldSerializeValue(object component)
        {
            return false;
        }
    }
}
